OfficeDev/microsoft-teams-library-js
# [Task8134448] Implementing the visualMedia API in teamsjs

Create a new API named visualMedia which will support image sub-capability.

## Description

The goal of these changes is to introduce a new API called the 'visualMedia API'. It includes an image sub-capability that allows the host app to upload and capture up to 10 images for the Mos Apps.

### Main changes in the PR:

1. Implemented the visualMedia API
2. Implemented the image sub-capability with two functions: captureImages() and uploadImages()

## Validation

### Validation performed:

1. Build teamsjs code and start-test-app
2. Build orange ios app (host app) and open local test app
4. Scroll to the Visual Media API section, click on "check visual media image capability" to make sure image sub-capability is supported by the host app.
5. Click on "request visual media permission" if the result is false from "has visual media permission"
6. Use the testing inputs:  
For captureImages: { "maxVisualMediaCount": 5, "sourceProps": { "source": 1, "cameraRestriction": 1 }, "commonImageProps": { "shouldAllowInkingOnImages": false, "shouldAllowTextStickersOnImages": false, "enableFilterOnImages": false } }
For retrieveImages: { "maxVisualMediaCount": 10, "sourceProps": {"source": 2}, "commonImageProps": { "shouldAllowInkingOnImages": false, "shouldAllowTextStickersOnImages": false, "enableFilterOnImages": false } }
8. The expected result (contains image information) showed in the text box

### Unit Tests added: 

>  Yes

### End-to-end tests added: 

> No

### Change file added:

> Yes

### Next/remaining steps:

- Back compt tests


### Screenshots:

<img width="468" alt="Screenshot 2023-10-24 at 9 15 39 PM" src="https://github.com/OfficeDev/microsoft-teams-library-js/assets/3358651/7d8942eb-da33-48c0-948f-a1e1a6f36e6b">
<img width="468" alt="Screenshot 2023-10-03 at 9 03 05 PM" src="https://github.com/OfficeDev/microsoft-teams-library-js/assets/3358651/690a22ad-b37c-4939-9cff-6bcb4d792f54">
<img width="550" alt="Screenshot 2023-10-25 at 12 01 34 PM" src="https://github.com/OfficeDev/microsoft-teams-library-js/assets/3358651/147da811-5991-4a16-95d3-cb06acc0780f">
<img width="535" alt="Screenshot 2023-10-25 at 11 58 57 AM" src="https://github.com/OfficeDev/microsoft-teams-library-js/assets/3358651/67a35d01-ab05-49d4-9a1b-f9cae8a2dd85">
